src/models/emotion_detection/hf_loader.py

```
@@ -197,16 +207,34 @@ def load_emotion_model_multi_source(
                 with open(archive_path, "wb") as f:
                     f.write(r.content)
-            # Extract
+            # Extract safely
             extract_dir = tempfile.mkdtemp(prefix="model_", dir=cache_dir)
             if archive_path.endswith(".tar.gz") or archive_path.endswith(".tgz"):
                 with tarfile.open(archive_path, "r:gz") as tar:
-                    tar.extractall(path=extract_dir)
+                    for member in tar.getmembers():
+                        # Validate member name to prevent path traversal
+                        if os.path.isabs(member.name) or ".." in member.name:
+                            raise ValueError(f"Unsafe archive member: {member.name}")
```

Ruff (0.13.0)

Abstract `raise` to an inner function

(TRY301)